conduit/src/Data/Conduit/Internal/Conduit.hs

```diff
@@ -877,6 +879,20 @@ runConduit :: Monad m => ConduitT () Void m r -> m r
 runConduit (ConduitT p) = runPipe $ injectLeftovers $ p Done
 {-# INLINE [0] runConduit #-}
 
+-- | Version of 'bracketP' for working with libraries that only expose an
+-- 'Acauire' based API.
+--
+-- @since 1.3.2
+bracketAcquireP :: MonadResource m
+                => Acquire a
+                -> (a -> ConduitT i o m r)
+                -> ConduitT i o m r
+bracketAcquireP acquire inside = ConduitT $ \rest -> do
```

Comment:
What's the reason for needing the underlying `ConduitT` representation here? Is it just for performance? I ask because I'm curious about what made `Acquire` difficult to use directly with the previous API.